### DISCUSSION TITLE
[@mantine/dates] DateInput does not close date picker or properly format the date on Enter key press

### DISCUSSION THREAD

**Discussion by ciacka** [mantinedev/mantine#7300] [Feature requests]
### Dependencies check up

- [X] I have verified that I use latest version of all @mantine/* packages

### What version of @mantine/* packages do you have in package.json?

7.11.1

### What package has an issue?

@mantine/dates

### What framework do you use?

Vite

### In which browsers you can reproduce the issue?

Chrome

### Describe the bug

When using the `DateInput` component, pressing the Enter key after typing a date does nothing. When the component is embedded inside a `<form>`, this creates a problem where the form gets submitted, but the date picker remains visible. Furthermore, if I then click on another DateInput, the previous date picker does not disappear (see attached video for reference).

https://github.com/user-attachments/assets/d0d13e04-bcd5-4abe-9de9-9403957478d0

It would be great if pressing the Enter key on the `DateInput` could:
	1.	Properly format the entered date.
	2.	Close the date picker.

### If possible, include a link to a codesandbox with a minimal reproduction

https://codesandbox.io/p/sandbox/mantine-date-input-on-enter-5zwcxn

### Self-service

- [ ] I would be willing to implement a fix for this issue

**Comment by roirh** [comment]
Hi, i'm interested in this feature too
Right now I'm capturing the onKeyDown event and forcing an onBlur event to close the dropdown
```
onKeyDown={(e) => {
    if (e.key === 'Enter') {
        console.log("Enter pressed")
        e.preventDefault()
        e.currentTarget.blur()
    }
}}
```

I can implement the change to be de default behaviour, it but i not sure if it has to be accepted first.
